src/Tasks/Microsoft.NET.Build.Extensions.Tasks/msbuildExtensions-ver/SolutionFile/ImportAfter/Microsoft.NET.Sdk.Solution.targets

```diff
@@ -37,4 +37,12 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <Target Name="_CheckForSolutionLevelTargetFramework"
+          BeforeTargets="Build;Publish;Clean;Store;VSTest;_MTPBuild"
```

Comment:
Note: for the case of `dotnet test --no-build -f ...`, this target isn't going to run.